packages/textkit/src/engines/linebreaker/bestFit.js

```diff
@@ -35,7 +35,25 @@ const getNextBreakpoint = (subnodes, widths, lineNumber) => {
       sum.shrink += node.shrink;
     }
 
-    if (sum.width - sum.shrink > lineLength) break;
+    if (sum.width - sum.shrink > lineLength) {
+      let j;
+      if (i === 0) {
+        // the first block is always "box" type
+        j = i + 1;
+      } else {
+        j = i;
+      }
+
+      while (
+        j < subnodes.length &&
+        (subnodes[j].type === 'glue' || subnodes[j].type === 'penalty')
+      ) {
+        j++;
+      }
+
+      position = j - 1;
```

Comment:
please run all this code only when position is `null`, like

```js

if (position == null) {
  // code
}

```

Reply:
Thanks!